### PR TITLE
Fee estimation on the wallet fullnode

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

File: Stratis.Bitcoin/Wallet/WalletManager.cs [hunk 1]
```diff
@@ -479,7 +479,7 @@ public ISecret GetKeyForAddress(string password, HdAddress address)
             }
 
             // calculate which addresses needs to be used as well as the fee to be charged
-            var calculationResult = this.CalculateFees(spendableTransactions, amount);
+            var calculationResult = this.CalculateFees(spendableTransactions, amount, 5);
```
Comment (bokobza) [inline Stratis.Bitcoin/Wallet/WalletManager.cs]: make this work with the feeType.
also, please make these ints constants.

File: Stratis.Bitcoin/Wallet/WalletManager.cs [hunk 2]
```diff
@@ -524,13 +524,15 @@ public ISecret GetKeyForAddress(string password, HdAddress address)
...
+        private (List<TransactionData> transactionsToUse, Money fee) CalculateFees(IEnumerable<TransactionData> spendableTransactions, Money amount, int targetConfirmations)
         {
-            // TODO make this a bit smarter!     
-            Money fee = new Money(new decimal(0.001), MoneyUnit.BTC);
+            Money fee = 0;
             List<TransactionData> transactionsToUse = new List<TransactionData>();
+            var inputCount = 1;
             foreach (var transaction in spendableTransactions)
             {
+                fee = this.walletFeePolicy.GetMinimumFee(inputCount * 180 + 74, targetConfirmations);
```
Comment (bokobza) [inline Stratis.Bitcoin/Wallet/WalletManager.cs]: please give some details about these numbers

**Comment by dangershony** [comment]
@bokobza this has been fixed

File: Stratis.Bitcoin.IntegrationTests/FeeTests.cs [hunk 3]
```diff
@@ -17,11 +19,16 @@ namespace Stratis.Bitcoin.IntegrationTests
     [TestClass]
     public class FeeTests
 	{
-		[TestMethod]
+	    public FeeTests()
```
Comment (bokobza) [inline Stratis.Bitcoin.IntegrationTests/FeeTests.cs]: not necessary anymore :-)

**Pull request merged**